### PR TITLE
Feat: Added possibility to create the Repository in the Builder

### PR DESCRIPTION
**Pull request by michael-schmetter**
# Description

Refactored the UnleashBuilder.php to enable it to also build only the repository

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [X] Unit tests
- [ ] Spec Tests
- [ ] Integration tests / Manual Tests

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by michael-schmetter** [comment]
@RikudouSage What do you think?

**Comment by RikudouSage** [comment]
@michael-schmetter Sorry, I totally missed this! I'll do a code review today and let you know.

File: src/UnleashBuilder.php [hunk 1]
```diff
@@ -376,7 +375,82 @@ public function withMetricsBucketSerializer(?MetricsBucketSerializer $metricsBuc
...
+
+        return new DefaultUnleashRepository($dependencyContainer->getHttpClient(), $dependencyContainer->getRequestFactory(), $dependencyContainer->getConfiguration());
+    }
+
     public function build(): Unleash
+    {
+        $dependencyContainer = $this->initializeContainerWithConfiguration();
+        assert($dependencyContainer->getConfiguration() !== null);
+
+        $repository = new DefaultUnleashRepository($dependencyContainer->getHttpClient(), $dependencyContainer->getRequestFactory(), $dependencyContainer->getConfiguration());
```
Comment (RikudouSage) [inline src/UnleashBuilder.php]: This should use the `buildRepository()` method to not duplicate logic. Though I don't like the idea of building the container twice like that, so perhaps a private method, that will build the repository and accepts nullable container as a parameter?

If you can think of other solutions, feel free to implement them.

**Review by RikudouSage — CHANGES_REQUESTED** [review]
Looks really good! I only found one small issue.

File: src/UnleashBuilder.php (hunk 1 above)
Comment (michael-schmetter) [reply src/UnleashBuilder.php]: @RikudouSage I extracted it to a private method `createRepository`.

Comment (michael-schmetter) [reply src/UnleashBuilder.php]: @RikudouSage Fine for you?

Comment (RikudouSage) [reply src/UnleashBuilder.php]: Yes, looks great, thanks!

**Review by RikudouSage — APPROVED** [review]

**Comment by michael-schmetter** [comment]
@RikudouSage Short question: What do you think when this will be released?

**Comment by RikudouSage** [comment]
@michael-schmetter Just released it.

**Comment by michael-schmetter** [comment]
> @michael-schmetter Just released it.

Any chance to get that to a PHP 8.2 compatible version?

**Pull request merged**